packages/transformers/js/src/JSTransformer.js

```diff
@@ -854,6 +854,23 @@ export default (new Transformer({
           dep.symbols.ensure();
           dep.symbols.set('*', '*', convertLoc(loc), true);
         }
+
+        if (
+          symbol_result.has_cjs_exports ||
+          (!symbol_result.is_esm &&
+            deps.size === 0 &&
+            symbol_result.exports.length === 0) ||
+          (symbol_result.should_wrap && !asset.symbols.hasExportSymbol('*'))
+        ) {
+          asset.symbols.set('*', `$`);
+        }
+
+        for (let d of deps.values()) {
+          if (d.priority === 'lazy') {
+            d.symbols.ensure();
+            d.symbols.set('*', '$');
+          }
+        }
```

Comment:
What were these extra symbols for? Maybe we should add a comment explaining them.

Reply:
I've added a comment. Without this, a side-effect-free dynamic import might be optimized away completely (because according to the transformer, no symbols are actually used from the import).